client/app/bundles/course/assessment/submission/actions/index.js

```diff
@@ -146,13 +146,15 @@ export function unsubmit(submissionId) {
       .then((response) => response.data)
       .then((data) => {
         dispatch({ type: actionTypes.UNSUBMIT_SUCCESS, payload: data });
+        dispatch(initiateAnswerFlagsForAnswers({ answers: data.answers }));
         dispatch(setNotification(translations.updateSuccess));
       })
       .catch((error) => {
         dispatch({ type: actionTypes.UNSUBMIT_FAILURE });
         dispatch(
           setNotification(translations.updateFailure, buildErrorMessage(error)),
         );
+        dispatch(resetExistingAnswerFlags());
```

Comment:
Do we need the reset here?

Reply:
after more thoughts given to this, it's actually not needed.
The reason is because when the unsubmit has failure, it won't convert the answer state from Submitted to Attempting, and hence the saving flag is not necessary to be handled here (can be left undefined)

Reply:
On another thought, should we reset the savingStatus after finalising the submission?

Reply:
for this one, if we don't handle the answerFlags in here, no need to handle the answerFlags in finalising, for the exact same reason. We don't need saving status if we cannot update the answer (which, I believe, is the traits of Submitted answer)

The behavior is that since we don't do anything regarding the savingStatus for finalising, the redux state will become undefined and hence it will impose the exact same behavior as if we reset it.

Reply:
Actually I am confused now. Say we have answer 1 and 2 and a finalised submission. When the submission page is first visited, the answerFlags for each answer is initialized to be none. So when unsubmitting, it should be maintained that way? Am I missing something here?

Reply:
Sorry, I made the wrong diagnosis for the root cause of this behavior.
The more complete reasoning behind this issue has been updated in the description.

Basically, the correct approach to fix this issue is to purge the old answerIds' answerFlag state and initialise the new ones.

As for finalising, we don't need to do anything, still, since we don't show any saving indicator while the answer is finalised.
When the un-submitting fails, we also don't need to do anything specifically for the answerFlags since it won't be used (we cannot modify the answer in the Submitted state)